wsi_superpixel_guided_labeling/web_client/views/vue/components/ActiveLearning/ActiveLearningContainer.vue

```diff
@@ -199,6 +205,37 @@ export default Vue.extend({
                 this.updateMapBoundsForSelection();
                 this.drawLabels();
             });
+            this.viewerWidget.on('g:drawOverlayAnnotation', (element, layer) => {
+                if (element.type === 'pixelmap') this.overlayLayer = layer;
+            });
+        },
+        updatePixelmapLayerStyle() {
+            _.forEach(this.overlayLayer.features(), (feature) => {
+                feature.style('color', (d, i) => {
+                    if (d < 0 || d >= this.categories.length) {
+                        console.warn(`No category found at index ${d} in the category map.`);
+                        return 'rgba(0, 0, 0, 0)';
+                    }
+                    const category = this.categories[d];
+                    return (i % 2 === 0) ? category.fillColor : category.strokeColor;
+                });
+            });
+            this.overlayLayer.draw();
+        },
+        saveAnnotations(imageId) {
+            // If we dont specify an image, save all images
+            const idsToSave = imageId ? [imageId] : Object.keys(this.annotationsByImageId);
+            this.backboneParent.saveLabelAnnotations(idsToSave);
+        },
+        synchronizeCategories() {
+            _.forEach(Object.values(this.annotationsByImageId), (annotations) => {
+                const superpixelElement = annotations.labels.get('annotation').elements[0];
+                if (superpixelElement) {
+                    superpixelElement.categories = JSON.parse(JSON.stringify(this.categories));
+                }
+            });
+            this.saveAnnotations(this.selectedImageId);
+            this.updatePixelmapLayerStyle();
```

Comment:
Could these functions be moved to the store? Since they're also defined in `ActiveLearningInitialLabels`. The store could probably be moved into the parent directory and shared between the two.

Reply:
Moved :+1:

Reply:
With the new support for the config file the only redundant function is now `updatePixelmapLayerStyle` which is still in the store. I've removed `synchronizeCategories` and `saveAnnotations` from the store.